From 913c100ff6dbc920733a017e1436eb0b6d1c302c Mon Sep 17 00:00:00 2001
From: Pierre Ossman <ossman@cendio.se>
Date: Wed, 25 Nov 2020 13:33:27 +0100
Subject: [PATCH 01/21] Better parsing errors

There are a lot of things that can go wrong, so let's try to be more
clear about what we don't like in the exports file.
---
 Config/exports.y | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/Config/exports.y b/Config/exports.y
index 4378884..f39cacf 100644
--- a/Config/exports.y
+++ b/Config/exports.y
@@ -323,6 +323,7 @@ static void set_hostname(const char *name)
 	struct hostent *ent;
 
 	if (strlen(name) + 1 > NFS_MAXPATHLEN) {
+		logmsg(LOG_CRIT, "hostname '%s' is too long", name);
 		e_error = TRUE;
 		return;
 	}
@@ -348,8 +349,10 @@ static void set_ipaddr(const char *addr)
 {
 	strcpy(cur_host.orig, addr);
 	
-	if (!inet_aton(addr, &cur_host.addr))
+	if (!inet_aton(addr, &cur_host.addr)) {
+		logmsg(LOG_CRIT, "could not parse address '%s'", addr);
 		e_error = TRUE;
+	}
 	cur_host.mask.s_addr = 0;
 	cur_host.mask.s_addr = ~cur_host.mask.s_addr;
 }
@@ -384,8 +387,10 @@ static void set_ipnet(char *addr, int new)
 	if (new)
 		cur_host.mask.s_addr = make_netmask(atoi(net));
 	else
-		if (!inet_aton(net, &cur_host.mask))
+		if (!inet_aton(net, &cur_host.mask)) {
+			logmsg(LOG_CRIT, "could not parse network mask '%s'", net);
 			e_error = TRUE;
+		}
 
 	*pos = '/';
 	strcpy(cur_host.orig, addr);
@@ -447,6 +452,8 @@ static void add_option_with_value(const char *opt, const char *val)
  */
 void yyerror(U(char *s))
 {
+	logmsg(LOG_CRIT, "parser error: %s", s);
+
 	e_error = TRUE;
 	return;
 }
-- 
2.42.0

